File: Java/labs/lab15_Command/task_3_1 – solution/src/Lamp.java
public class Lamp {

    final private String name;
    private boolean isLightOn;

    public Lamp(String name) {
        this.name = name;
        this.isLightOn = false;
    }

    public Lamp() {
        this("default");
    }

    public void lightOn() {
        if (isLightOn) {
            return;
        }
        System.out.println(name + ": Light is on");
        isLightOn = true;
    }

    public void lightOff() {
        if (!isLightOn) {
            return;
        }
        System.out.println(name + ": Light is off");
        isLightOn = false;
    }
}

File: Cpp/lab15_Command/task_3_1/src/lamp.cpp
#include "lamp.hpp"

#include <iostream>


Lamp::Lamp(std::string const& name)
: name(name), isLightOn(false) {}

Lamp::Lamp()
: Lamp("default") {}

void Lamp::lightOn()
{
  if (this->isLightOn)
  {
    return;
  }

  std::cout << name << ": Light is on" << std::endl;
  this->isLightOn = true;
}

void Lamp::lightOff()
{
  if (!this->isLightOn)
  {
    return;
  }

  std::cout << name << ": Light is off" << std::endl;
  this->isLightOn = false;
}
